syne_tune/optimizer/schedulers/searchers/searcher.py

```diff
@@ -551,17 +568,139 @@ def __init__(
+            # minimum of default value DEFAULT_NSAMPLE and the interger range is used.
+            if isinstance(hp_range, Integer):
+                if hp in self.num_samples:
+                    if self.num_samples[hp] > len(hp_range):
+                        self.num_samples[hp] = min(len(hp_range), DEFAULT_NSAMPLE)
+                        logger.info(
+                            'number of samples for "{}" is larger than its range. We set it to the minimum of the default number of samples (i.e. {}) and its range length (i.e. {}).'.format(
+                                hp, DEFAULT_NSAMPLE, len(hp_range)
+                            )
+                        )
+                else:
+                    self.num_samples[hp] = min(len(hp_range), DEFAULT_NSAMPLE)
+            # num_samples is ignored for categorical hps.
+            if isinstance(hp_range, Categorical) or isinstance(hp_range, FiniteRange):
+                if hp in self.num_samples:
+                    logger.info(
+                        'number of samples for categorical variable "{}" is ignored.'.format(
+                            hp
+                        )
+                    )
+
+    def _remove_duplicate_candidates(self, all_candidates) -> List[dict]:
+        """
+        Excludes _points_to_evaluate from all_candidates list, so that no duplicate
+        candidates exist.
+        Args:
+            all_candidates:  all candidates on grid generated by _generate_all_candidates_on_grid()
+
+        Returns:
+            de-duplicated list of candidates on grid
+        """
+        excl_list = ExclusionList.empty_list(self._hp_ranges)
+        for candidate in self._points_to_evaluate:
+            excl_list.add(candidate)
+
+        remaining_candidates = []
+        for candidate in all_candidates:
+            if not excl_list.contains(candidate):
+                remaining_candidates.append(candidate)
+                excl_list.add(candidate)
+
+        self._grid_candidates = remaining_candidates
+        self._candidate_index = 0
+        return
+
+    def _generate_all_candidates_on_grid(self) -> List[dict]:
+        """
+        Generates all configurations to be evaluated by placing a regular, equally spaced grid over the configuration space
+        Returns:
+            The set of all configurations on grid.
+        """
+        hp_keys = []
+        hp_values = []
+        ## adding categorical, finiteRange, scalar parameters
+        for hp, hp_range in reversed(list(self.config_space.items())):
+            if isinstance(hp_range, Float) or isinstance(hp_range, Integer):
+                continue
+            if isinstance(hp_range, Categorical):
+                values = hp_range.categories
+            elif isinstance(hp_range, FiniteRange):
+                values = hp_range.values
+            elif not isinstance(hp_range, Domain):
+                values = [hp_range]
+            hp_keys.append(hp)
```

Comment:
You were right rejecting my ask, please revert to old code. Some weirdo may call this with a non-supported Domain subtype.